run/convert.go

```diff
@@ -143,67 +142,66 @@ func mapSchemaForeignKeyReferences(isch *database.Schema, sch *data.Schema) erro
+			if column.IsFK {
+				refTable, ok := sch.TablesByName[c.ForeignKey.ForeignTableName]
 				if !ok {
 					log.Printf("Unmapped foreign table %v in %v", c.ForeignKey.ForeignTableName, isch.Name)
 					continue
 				}
-				foreignColumn, ok := foreignTable.ColumnsByName[c.ForeignKey.ForeignColumnName]
+				refColumn, ok := refTable.ColumnsByName[c.ForeignKey.ForeignColumnName]
 				if !ok {
 					log.Printf("Unmapped foreign column %v in %v.%v", c.ForeignKey.ForeignColumnName, isch.Name, c.ForeignKey.ForeignTableName)
 					continue
 				}
 
-				fc := &data.ForeignColumn{
-					Name:                     c.ForeignKey.Name,
-					ColumnName:               column.Name,
-					ForeignColumnName:        column.Name,
-					UniqueConstraintPosition: c.ForeignKey.UniqueConstraintPosition,
-					Column:        column,
-					ForeignColumn: foreignColumn,
+				fkColumn := &data.ForeignKeyColumn{
+					DBName:          c.ForeignKey.Name,
+					ColumnDBName:    column.DBName,
+					RefColumnDBName: refColumn.DBName,
+					Column:          column,
+					RefColumn:       refColumn,
 				}
-				column.ForeignColumn = fc
+				column.FKColumn = fkColumn
 
-				foreignColumn.IsForeignKeyReference = true
-				foreignColumn.ForeignKeyReferences = append(foreignColumn.ForeignKeyReferences, c.ForeignKey.Name)
-				foreignColumn.ForeignColumnsByForeignKeyReference[c.ForeignKey.Name] = column.ForeignColumn
+				refColumn.HasFKRef = true
+				refColumn.FKColumnRefs = append(refColumn.FKColumnRefs, fkColumn)
+				refColumn.FKColumnRefsByName[fkColumn.DBName] = fkColumn
 
-				if _, ok := table.ForeignColumnsByForeignKey[fc.Name]; !ok {
-					table.ForeignColumnsByForeignKey[fc.Name] = []*data.ForeignColumn{}
-				}
-				table.ForeignColumnsByForeignKey[fc.Name] = append(table.ForeignColumnsByForeignKey[fc.Name], fc)
-
-				if _, ok := distinctForeignColumnsByForeignKeyName[fc.Name]; !ok {
-					distinctForeignColumnsByForeignKeyName[fc.Name] = fc
+				if _, ok := fkColumnsByFKNames[fkColumn.DBName]; !ok {
+					fkColumnsByFKNames[fkColumn.DBName] = data.ForeignKeyColumns{fkColumn}
+				} else {
+					fkColumnsByFKNames[fkColumn.DBName] = append(fkColumnsByFKNames[fkColumn.DBName], fkColumn)
 				}
 			}
 		}
 
-		for _, fc := range distinctForeignColumnsByForeignKeyName {
-			mapForeignTable(fc)
+		for _, fkc := range fkColumnsByFKNames {
+			mapForeignTable(fkc)
 		}
 	}
 
 	return nil
 }
 
-func mapForeignTable(fc *data.ForeignColumn) {
-	table := fc.Column.Table
-	fTable := fc.ForeignColumn.Table
-	ft := &data.ForeignTable{
-		Name:             fc.Name,
-		TableName:        table.Name,
-		ForeignTableName: fTable.Name,
-		Table:            table,
-		ForeignTable:     fTable,
+func mapForeignTable(fkc data.ForeignKeyColumns) {
+	if len(fkc) == 0 {
+		return
 	}
 
-	table.ForeignKeys = append(table.ForeignKeys, fc.Name)
-	if _, ok := table.ForeignTablesByForeignKey[fc.Name]; !ok {
-		table.ForeignTablesByForeignKey[fc.Name] = ft
+	table := fkc[0].Column.Table
```

Comment:
can you put a comment here that explains that all the tables should be the same, so we can just grab the one off the first column?  It'll save someone some head scratching down the road.